src/mame/misc/toysoldier.cpp

```diff
@@ -0,0 +1,169 @@
+   
+PORT_START("IN1") //  Claw input 
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+    PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_1)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_2)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_3)
+ 	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+  
+PORT_START("IN2")
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT)
+    PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_JOYSTICK_UP)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_JOYSTICK_DOWN)
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_R)
+    PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_BUTTON1) 
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_W)
+    PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_KEYPAD) PORT_CODE(KEYCODE_Q)
+
+PORT_START("IN3") // coin
+	PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_COIN2)
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_COIN1)
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+
+ 
+PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+
+
+PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW2:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW2:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW2:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW2:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW2:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW2:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW2:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW2:8")
+INPUT_PORTS_END
+
+void toysoldier_state::program_map(address_map &map)
+{
+	map(0x0000, 0xffff).rom();
+}
+
+void toysoldier_state::io_map(address_map &map)
+{
+	map(0x8000, 0x8001).w("opll", FUNC(ym2413_device::write));
+	map(0x9000, 0x9001).rw("ay", FUNC(ay8910_device::data_r), FUNC(ay8910_device::address_data_w));
+    map(0xa000, 0xa000).portr("IN1"); //  claw up claw down return input
+//	map(0xa000, 0xa000).nopw(); // ???
+//  map(0xb000, 0xb000).nopw(); //  aplus ic
+    map(0xb000, 0xb000).portr("IN2"); // joystick
+//	map(0xc000, 0xc000).nopw(); // ???
+    map(0xc000, 0xc000).portr("IN3");  // Coin
+//  map(0xd000, 0xd000).unmapw(); // 7 segment display  control
+    map(0xe000, 0xe000).nopw(); // Super Card
+}
+
+void toysoldier_state::machine_start()
+{
+}
+
+void toysoldier_state::toysoldier(machine_config &config)
+{
+	/* basic machine hardware */
```

Comment:
Please use C++ style comments for one-liners
// basic machine hardware
instead of
/* basic machine hardware */